java/sage/MediaServerRemuxer.java

```diff
@@ -0,0 +1,1008 @@
+
+    transferBuffer = new byte[initData];
+    this.fileChannel = fileChannel;
+    writer = new RemuxWriter();
+
+    remuxer2 = MPEGParser2.openRemuxer(inputFormat, outputFormat, streamFormat, subFormat, tuneStringType, channel, tsid, data1, data2, data3, writer);
+
+    if (remuxer2 == null)
+    {
+      throw new IllegalArgumentException("Unable to initialize the remuxer.");
+    }
+
+    remuxerMap.put(currentFile, this);
+  }
+
+
+  public void setBufferLimit(long bufferLimit)
+  {
+    this.bufferLimit = bufferLimit;
+  }
+
+  /**
+   * Get if the remuxer is initialized.
+   * <p/>
+   * This essentially also means data is being written out which can be verified by checking the
+   * size.
+   *
+   * @return If the remuxer is currently initialized.
+   */
+  public boolean isInitialized()
+  {
+    return containerFormat != null;
+  }
+
+  /**
+   * Get the current container format.
+   * <p/>
+   * This will return null if the container format has not been detected.
+   *
+   * @return The container format.
+   */
+  public ContainerFormat getContainerFormat()
+  {
+    return containerFormat;
+  }
+
+  /**
+   * Get the current file size.
+   * <p/>
+   * This is a thread-safe method.
+   *
+   * @return The current file size.
+   */
+  public long getFileSize()
+  {
+    return fileSize.get();
+  }
+
+  /**
+   * Get the current remux output mode.
+   *
+   * @return The selected remux output mode.
+   */
+  public int getOutputFormat()
+  {
+    return outputFormat;
+  }
+
+  public boolean isClosed()
+  {
+    return closed;
+  }
+
+  /**
+   * Close the remuxer instance and block further writing.
+   */
+  public void close()
+  {
+    if (closed)
+      return;
+
+    if (currentFile != null)
+      remuxerMap.remove(currentFile);
+
+    if (remuxer2 != null)
+    {
+      remuxer2.close();
+      remuxer2 = null;
+
+      try
+      {
+        if (partialTransferIndex > 0)
+          writer.write(partialTransfer, 0, partialTransferIndex);
+
+        writer.flush();
+      }
+      catch (IOException e)
+      {}
+    }
+
+    closed = true;
+  }
+
+  /**
+   * Checks the currently buffered write data to see if we can safely split it anywhere.
+   * <p/>
+   * If the data can be split, the data up to the split will be written out leaving the data that
+   * should be written to the new file still in the buffer.
+   * <p/>
+   * Calling this method also makes the remuxer aware that we are trying to switch and will check
+   * the bytes on each write for an opportunity. Once a switching point is detected, all new data
+   * will be buffered until switchOutput is called.
+   *
+   * @param filename The new filename to switch to.
+   * @param uploadId The upload ID to be used to authorize the new filename.
+   */
+  public void startSwitch(String filename, int uploadId)
+  {
+    synchronized (switchLock)
+    {
+      // Don't allow SWITCH to the same file.
+      if (!switching && new File(filename) != currentFile)
```

Comment:
The second comparison here will never be equal because you are constructing a new File object so it's object reference cannot be the same as an existing one. You should just compare the filenames as String values instead.